sh_scrapy/crawl.py

```diff
@@ -155,6 +155,27 @@ def _launch():
     _run_usercode(job['spider'], args, _get_apisettings, loghdlr)
 
 
+def list():
+    """ An entrypoint for list-spiders."""
+    try:
+        from scrapy.exceptions import ScrapyDeprecationWarning
+        warnings.filterwarnings(
+            'ignore', category=ScrapyDeprecationWarning, module='^sh_scrapy')
+        from sh_scrapy.env import get_scrapy_list_args_and_env, decode_uri
+        job = decode_uri(envvar='JOB_DATA')
+        assert job, 'JOB_DATA must be set'
+        args, env = get_scrapy_list_args_and_env(job)
+        os.environ.update(env)
+
+        from sh_scrapy.env import setup_environment
+        setup_environment()
+    except:
+        _fatalerror()
+        raise
+
+    _run_usercode(None, args, _get_apisettings)
+
```

Comment:
We don't change sys IO when getting a spiders list, so no need to do it in the same way like it's done for `crawl` logic.